api/xds/gloo-discovery-service.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
```

Comment:
This proto is essentially the same as the Envoy ADS proto. I renamed it since we are only using it for ext-auth and rate-limiting communication internally and aren't directly interacting with Envoy